jac-uk/admin
# #1666 Stages and statuses

## What's included?

Exercise > Stages has been re-written to include the following functionality for all stages:
 - Search. Now included on all stages
 - Filters. Now includes options for current stage statuses and previous stage success statuses
 - Page size. Can now be changed up to 500 rows per page
 - EMP can be changed without needing to select a new status
 - Update status. Now includes back navigation
 - Move Back. Now includes back navigation

In addition we have removed old QT code and added a number of improvements and fixes including:
- [x] Remove QT code
- [x] Include `npm run local:serve` package script so developers can more easily run this app against the local emulator
- [x] Update header to display LOCAL when using local emulator (was displaying DEVELOP)
- [x] Tasks navigation should include a DONE tag so user knows which tasks have been completed
- [x] Sift, Scenario and Selection Day panel packs should include the correct candidates:
  - Sift = review/applied stage
  - Scenario = review/applied stage and 'passed first test' status
  - Selection Day = selected stage
- [x] Bugfix: Include Sift in Reports navigation (was previously showing for Name-blind paper sift but not for Paper sift)
- [x] Bugfix: App and service offline page should have a favicon
- [x] Bugfix: Timeline. IA return date should not include time
- [x] Bugfix: Create Exercise. Add more info check-boxes should be required
- [x] Bugfix: Dashboard. Only show report elements if we have the data
- [x] Bugfix: Character Issues report uses correct helper to retrieve stage and status dropdowns
- [x] Bugfix: Custom report uses correct helper to retrieve stage and status dropdowns
- [x] Bugfix: Reasonable adjustments report uses correct helper to retrieve stage and status dropdowns

Closes #1666 
Closes #1933 

_**Note that this PR also includes code to support a new set of Stages and Statuses. These will be tested on a per exercise basis outside of this PR. The key focus here is to make sure existing functionality still works as expected**_

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?


Primarily we are testing that the Stages part of admin still works as expected and includes some improved functionality.

Set up or locate an exercise with applications
[example 1: a few applications](https://jac-admin-develop--pr1931-feature-1861-process-e5nom497.web.app/exercise/euHYlnsu1DdSlSkve2BW/stages/review)
[example 2: lots of applications](https://jac-admin-develop--pr1931-feature-1861-process-e5nom497.web.app/exercise/4xP8RY7GeoaS1yKqYJLw/stages/review)
Then navigate to Stages and try the following:

**List view**
- Search for an application/candidate
- Filter the applications
- Change page size

**Update status**
Select application(s) from the stage and click the Update Status button:
- Check that the correct status options are available to choose from
- Note that in Review stage the status options should only include relevant statuses (e.g. If a sift is not selected then it will not include Passed/Failed Sift options. Try making changes to the shortlisting methods)
- Try making a status change
- Try updating EMP and status
- Try updating EMP without selecting a new status
- Try going back to the list without making any changes

**Move back**
- Try going back to the list without making any changes
- Try moving application(s) back to the previous stage


## Risk - how likely is this to impact other areas?
🔴 High risk - this includes a lot of changes to shared code

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
